File: CS50x/sentimental-readability/readability.py
from cs50 import get_string


def get_text():
    text = get_string("Text: ")
    return text


def main():
    # import user text
    text = get_text()

    # count amount of words, sentences etc
    words = 1
    sentences = 0
    letters = 0

    # symbols to determine sentences
    symbols = ["!", "?", "."]

    text_length = len(text)
    for i in range(text_length):
        if text[i] in symbols:
            sentences += 1
        elif text[i] == " ":
            words += 1

    # count amount of letters in text
    letters = text_length - sentences - words

    # formula for grade reading level: 0.0588 * L - 0.296 * S - 15.8
    # L = avg number of letters per 100 words
    # S = number of sentences per 100 words text
    L = (letters / words) * 100
    S = (sentences / words) * 100
    grade = 0.058 * L - 0.296 * S - 15.8
    grade = int(round(grade))

    print(f"Rounded grade: {grade}")
    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(grade)}")


main()

File: CS50x/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

void calculate_level(string text);

int main(void)
{
    // ask user to input text
    string text = get_string("Text: ");

    // call function to determine reading level
    calculate_level(text);
}

void calculate_level(string text)
{
    int letters = 0;
    int sentences = 0;
    int words = 1;
    int location = 0;

    // Count amount of letters, words and sentences
    while (text[location] != 0)
    {
        // if character is a-z count towards letters
        if (tolower(text[location]) >= 97 && tolower(text[location]) <= 122)
        {
            letters++;
            location++;
        }
        // if character is punctuation (!?.) count towards sentences
        else if (text[location] == 33 || text[location] == 63 || text[location] == 46)
        {
            sentences++;
            location++;
        }
        // count spaces between characters to count words
        else if (text[location] == 32)
        {
            words++;
            location++;
        }
        else
        {
            location++;
        }
    }

    // index = 0.0588 * L - 0.296 * S - 15.8
    // L = avg letters per 100 words, S = avg sentences per 100 words

    float L = (letters / (float)words) * 100;
    float S = (sentences / (float)words) * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;
    int rating = round(index);

    if (rating < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (rating > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", rating);
    }
}